mesonbuild/backend/xcodebackend.py

```diff
@@ -751,6 +753,7 @@ def generate_xc_build_configuration(self):
                     args = pargs + gargs + targs
                     if args:
                         langargs[langnamemap[lang]] = args
+                langargs['C'] += cargs
```

Comment:
Should this really go into `langargs['C']` and not into `langargs[langnamemap[lang]]`?

Sorry I just noticed this.

Reply:
we probably want to set these flags for C and C++.  I think the main concern is not to set these flags for other languages like objective-c, etc.

Reply:
https://developer.apple.com/library/archive/documentation/DeveloperTools/Reference/XcodeBuildSettingRef/1-Build_Setting_Reference/build_setting_ref.html

xcode spec has "OTHER_CFLAGS" and "OTHER_CPLUSPLUSFLAGS".

Default value of "OTHER_CPLUSPLUSFLAGS" is "OTHER_CFLAGS" so these flags should apply for C and C++

Reply:
okay, That's probably fine for now then. If we run into a case where we need to apply them to ObjC we can handle that later.